bpython/curtsiesfrontend/repl.py

```diff
@@ -179,12 +179,19 @@ def add_input_character(self, e: str) -> None:
         )
         self.cursor_offset += 1
 
-    def readline(self):
-        self.has_focus = True
-        self.repl.send_to_stdin(self.current_line)
-        value = self.coderunner.request_from_main_context()
-        self.readline_results.append(value)
-        return value
+    def readline(self, size=-1):
```

Comment:
While at it, could you also add type annotations here? That would be `def readline(self, size: int =  -1) -> str`.